### ISSUE TITLE
Docs: Advanced Types section "Interfaces vs. Type Aliases" is contradictory

### ISSUE THREAD

**Issue by carnesen** [Microsoft/TypeScript#28980]
**TypeScript Version:**  N/A

**Search Terms:**
label: Docs "Advanced Types" "Interfaces vs Type Aliases"

I've been trying to understand the differences between interfaces and type aliases, and I've found contradictory information on the ["Advanced Types"](https://www.typescriptlang.org/docs/handbook/advanced-types.html) page of the docs. The "Type Aliases" section begins with "Type aliases create a new name for a type." Just a few paragraphs later though in the "Interfaces vs. Type Aliases" subsection, it says: "Type aliases **don’t** create a new name". (Emphasis is mine). See also [this blog post](https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c), which enumerates some other aspects of this section that would benefit from clarification/update.

**Comment by pawelgrzybek** [comment]
I agree with @carnesen. Another contradictory info it this sentence:

> In the code below, hovering over interfaced in an editor will show that it returns an Interface, but will show that aliased returns object literal type.

```ts
type Alias = { num: number }
interface Interface {
    num: number;
}
declare function aliased(arg: Alias): Alias;
declare function interfaced(arg: Interface): Interface;
```

It shows a correct type for `interfaced` which is `Interface`. Documentation says that for `aliased` function the returned type is object literal but this is not the case — my code editor shows me `Alias` as a returned type.

![28980](https://user-images.githubusercontent.com/3697104/55743406-20630f80-5a2a-11e9-986b-4f9bd2819427.jpg)

**Comment by behnood-eghbali** [comment]
"An interface can be used in an extends or implements clause,
which means you can explicitly use them when defining other interfaces and classes. An interface can also
accept type arguments, making the interface generic. A type alias can do neither of these." - Pro TypeScript Book

**Comment by behnood-eghbali** [comment]
"An interface can be used in an extends or implements clause, which means you can explicitly use them when defining other interfaces and classes. An interface can also accept type arguments, making the interface generic. A type alias can do neither of these." - Pro TypeScript Book

**Comment by pawelgrzybek** [comment]
Hi @behnood-eghbali. Second part of this de is outdated too. This part of docs is a great candidate to update after 3 years. Things changed.
